### PR TITLE
feat: upgrade contracts to vrf v2

### PR DESCRIPTION
**Pull request by petia-latinova**

### CODE REVIEW

File: packages/frontend/components/vrf/RandomNFT/ExternalLink.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export interface Props {
 export function ExternalLink({ tokenId }: Props): JSX.Element {
   const { chainId } = useEthers()
 
-  const contract = useContractConfig('RandomSVG')
+  const contract = useContractConfig('RandomSVGV2')
```
Comment (imollov) [inline packages/frontend/components/vrf/RandomNFT/ExternalLink.tsx:22]: Please make the changes directly on `RandomSVG`. We don't need to keep the two versions.

File: packages/hardhat/deploy/05_Deploy_RandomSVG.ts [hunk 2]
```diff
@@ -10,25 +17,91 @@ const func: DeployFunction = async function ({
...
+    await deployToHardhatNetwork(deploy, get, deployer)
   } else {
-    linkTokenAddress = networkConfig[chainId].linkToken as string
-    vrfCoordinatorAddress = networkConfig[chainId].vrfCoordinator as string
+    await deployToNonHardhatNetwork(deploy, deployer, chainId)
   }
-  const { keyHash, fee } = networkConfig[chainId]
+}
+
+async function deployToHardhatNetwork(
```
Comment (imollov) [inline packages/hardhat/deploy/05_Deploy_RandomSVG.ts:28]: `deployToLocalNetwork`?

File: packages/hardhat/deploy/05_Deploy_RandomSVG.ts [hunk 3]
```diff
@@ -10,25 +17,91 @@ const func: DeployFunction = async function ({
...
+  )
+  if (!consumerIsAdded) {
+    await vrfCoordinatorV2Mock?.addConsumer(
+      vrfSubscriptionId,
+      RandomSVG.address
+    )
+  }
+}
+
+async function deployToNonHardhatNetwork(
```
Comment (imollov) [inline packages/hardhat/deploy/05_Deploy_RandomSVG.ts:71]: `deployToPublicNetwork`?

File: packages/hardhat/deploy/05_Deploy_RandomSVG.ts [hunk 4]
```diff
@@ -10,25 +17,91 @@ const func: DeployFunction = async function ({
...
+    args: [
+      vrfCoordinatorV2,
+      vrfSubscriptionId,
+      vrfGasLane,
+      vrfCallbackGasLimit,
+    ],
     log: true,
   })
+
+  const RandomSVG = await get('RandomSVG')
```
Comment (imollov) [inline packages/hardhat/deploy/05_Deploy_RandomSVG.ts:57]: Isn't the instance returned by the `deploy` function? Seems unnecessary to `get` it.

File: packages/hardhat/deploy/05_Deploy_RandomSVG.ts [hunk 5]
```diff
@@ -10,25 +17,91 @@ const func: DeployFunction = async function ({
...
+
+  const consumerIsAdded = await vrfCoordinatorV2Mock.consumerIsAdded(
+    vrfSubscriptionId,
+    RandomSVG.address
+  )
+  if (!consumerIsAdded) {
```
Comment (imollov) [inline packages/hardhat/deploy/05_Deploy_RandomSVG.ts:63]: In which case consumer is already added when running this script?

File: packages/hardhat/hardhat.config.ts [hunk 6]
```diff
@@ -70,6 +70,9 @@ const config: HardhatUserConfig = {
   },
   solidity: {
     compilers: [
+      {
+        version: '0.8.4',
+      },
```
Comment (imollov) [inline packages/hardhat/hardhat.config.ts:75]: Is it mandatory to use the new version?

File: packages/hardhat/deploy/05_Deploy_RandomSVG.ts (hunk 5 above)
Comment (petia-latinova) [reply packages/hardhat/deploy/05_Deploy_RandomSVG.ts:63]: This code is triggered not only during the deployment but also when the deployed script runs just to deploy some changes in other contract and we want to skip it for that situations and the chainlink mock contract doesn't work if you add the same consumer twice.

**Comment by imollov** [comment]
@ivaylonikolov7 pls rebase and squash into meaningful commits.

**Review by imollov — APPROVED** [review]

**Pull request merged**